js/content/common.js

```diff
@@ -212,6 +212,18 @@ let ExtensionLayer = (function() {
         script.parentNode.removeChild(script);
     };
 
+    // NOTE: use cautiously!
+    // Load remote script in the context of the current tab
+    self.loadInPageContext = function(fun) {
```

Comment:
If the major use case is going to chain calls to this, you might as well make it accept an array of urls as its first parameter and return a Promise that resolves when they are all loaded.

Reply:
The implementation of that would not be so clean I think.

Reply:
```js
function(srcs, sync=false) {
  function _makeScript(src) {
    return new Promise(function(resolve, reject) {
            let script = document.createElement("script");
            script.src = src;
            script.onload = resolve;
            script.onerror = reject;
            document.documentElement.appendChild(script);
            script.parentNode.removeChild(script);
        });
  }
  if (!Array.isArray(srcs)) {
    srcs = [srcs,];
  }
  let promise = sync ? Promise.resolve(true) : [];
  for (let src of srcs) {
    // domain check
    if (sync) {
      promise = promise.then(function() { return _makeScript(src); });
    } else {
      promise.push(_makeScript(src));
    }
  }
  return promise;
}
```

Reply:
`// domain check` how would I reject here if the domain isn't whitelisted?

Reply:
Done